synapseclient/models/wiki.py

```diff
@@ -1 +1,1018 @@
+        # Get the cache directory path to save the newly created gzipped file
+        cache_dir = os.path.join(synapse_client.cache.cache_root_dir, "wiki_content")
+        if not os.path.exists(cache_dir):
+            os.makedirs(cache_dir)
+
+        # Check if markdown looks like a file path and exists
+        if os.path.isfile(wiki_content):
+            # If it's already a gzipped file, save a copy to the cache
+            if wiki_content.endswith(".gz"):
+                file_path = os.path.join(cache_dir, os.path.basename(wiki_content))
+                shutil.copyfile(wiki_content, file_path)
+            else:
+                # If it's a regular html or markdown file, compress it
+                with open(wiki_content, "rb") as f_in:
+                    # Open the output gzip file
+                    file_path = os.path.join(
+                        cache_dir, os.path.basename(wiki_content) + ".gz"
+                    )
+                    with gzip.open(file_path, "wb") as f_out:
+                        f_out.writelines(f_in)
+
+        else:
+            # If it's a plain text, write it to a gzipped file and save it in the synapse cache
+            file_path = os.path.join(cache_dir, f"wiki_markdown_{self.title}.md.gz")
+            with gzip.open(file_path, "wt", encoding="utf-8") as f_out:
+                f_out.write(wiki_content)
+
+        return file_path
+
+    @staticmethod
+    def _get_file_size(filehandle_dict: dict, file_name: str) -> str:
+        """Get the file name from the response headers.
+        Arguments:
+            response: The response from the REST API.
+        Returns:
+            The file name.
+        """
+        filehandle_dict = filehandle_dict["list"]
+        available_files = [filehandle["fileName"] for filehandle in filehandle_dict]
+        # locate the contentSize for given file_name
+        for filehandle in filehandle_dict:
+            if filehandle["fileName"] == file_name:
+                return filehandle["contentSize"]
+        raise ValueError(
+            f"File {file_name} not found in filehandle_dict. Available files: {available_files}"
+        )
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Store the wiki page: {self.owner_id}"
+    )
+    async def store_async(
+        self,
+        *,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "WikiPage":
+        """Store the wiki page. If there is no wiki page, a new wiki page will be created.
+        If the wiki page already exists, it will be updated.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                    `Synapse.allow_client_caching(False)` this will use the last created
+                    instance from the Synapse class constructor.
+
+        Returns:
+            The created wiki page.
+
+        Raises:
+            ValueError: If owner_id is not provided or if required fields are missing.
```

Comment:
Each of the method for both the async and non-async functions should have at least 1 example to show how to use and call the function, along with the required imports. Ideally one could copy/paste the code, minimally change something, and run it.

Reply:
On it